38-allowlist-management.md

```diff
@@ -0,0 +1,162 @@
+# enhancement-38: Allowlist management
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Notes/Constraints/Caveats (optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [ ] Enhancement issue in release milestone, which links to pull request in [keylime/enhancements]
+- [ ] Core members have approved the issue with the label `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place
+- [ ] User-facing documentation has been created in [keylime/keylime-docs]
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+## Summary
+
+Allowlist management is, obviously, managing allowlists. Currently the allowlist
+is binded with agent records, there are two issues in current design:
+
+* Allowlist reusing: An allowlist can't be shared between agents, so we can't apply
+  the same data to identical environments. By support reusing, we also can provide
+  batch allowlist updates for matching agents.
+* Performance: Since the ima measurements could be very large, it has impact on query
+  efficiency when the allowlist does not participate in the agent processing, e.g., we
+  don't need to load the allowlist for updating agent state.
+
+
+## Motivation
+
+### Goals
+
+The goal is to provide allowlist management API and CLI for allowlist management,
+establish relationship between an allowlist and an agent. The allowlist used by agent
```

Comment:
Minor nit: Should be something like "and establish a relationship"
Minor nit: Should be "used by an agent"